consensus/istanbul/backend/engine.go

```diff
@@ -560,10 +561,53 @@ func (sb *Backend) SetChain(chain consensus.ChainReader, currentBlock func() *ty
 	sb.chain = chain
 	sb.currentBlock = currentBlock
 	sb.stateAt = stateAt
+
+	if bc, ok := chain.(*ethCore.BlockChain); ok {
+		go sb.newChainHeadLoop(bc)
+		go sb.updateReplicaStateLoop(bc)
+	}
+
 }
 
-// StartValidating implements consensus.Istanbul.StartValidating
-func (sb *Backend) StartValidating(hasBadBlock func(common.Hash) bool,
+func (sb *Backend) newChainHeadLoop(bc *ethCore.BlockChain) {
```

Comment:
I'll add the comment that it can be batched. The couple tests I did showed that I think it's much closer to real time when actively validating, and ran more into issues when syncing as a replica (problems with start, not with stop).